building guide

Well is pretty difficult to build this project without assistance, found casually another issue with this commands
```
cd dpc_linux
qmake -makefile DrevoPowerConsole.pro
make
sudo cp udev/77-drevo-usb-allow-wheel.rules /usr/lib/udev/rules.d/
```
But it gives me "CONNECT YOUR DEVICE"

PS
I was expecting more from a good keyboard, coming from this [site](https://www.drevo.net/dpc).

Hi there, 
I have found the repo from the [official Drevo page](https://www.drevo.net/dpc) and I have followed the instruction at the top of this issue but I got some errors. 
I work on a machine with Ubuntu 16.04 and I had installed libusb-1.0-0-dev from `apt-get` 

That's what I get after the `make` instruction.
```
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o main.o main.cpp
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o mainwindow.o mainwindow.cpp
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o qkeyboardctrl.o qkeyboardctrl.cpp
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o qkeyboardctrl_init.o qkeyboardctrl_init.cpp
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o customkeywidget.o customkeywidget.cpp
customkeywidget.cpp: In member function ‘void CustomKeyWidget::on_pushButton_save_clicked()’:
customkeywidget.cpp:132:22: warning: missing initializer for member ‘KeyData::profile’ [-Wmissing-field-initializers]
     KeyData data = {0};
                      ^
customkeywidget.cpp:132:22: warning: missing initializer for member ‘KeyData::key_value’ [-Wmissing-field-initializers]
customkeywidget.cpp:132:22: warning: missing initializer for member ‘KeyData::macro_type’ [-Wmissing-field-initializers]
customkeywidget.cpp:132:22: warning: missing initializer for member ‘KeyData::macro_value’ [-Wmissing-field-initializers]
customkeywidget.cpp:132:22: warning: missing initializer for member ‘KeyData::macro_value1’ [-Wmissing-field-initializers]
customkeywidget.cpp:132:22: warning: missing initializer for member ‘KeyData::macro_value2’ [-Wmissing-field-initializers]
customkeywidget.cpp:132:22: warning: missing initializer for member ‘KeyData::macro_name’ [-Wmissing-field-initializers]
customkeywidget.cpp: In member function ‘void CustomKeyWidget::on_pushButton_cancel_clicked()’:
customkeywidget.cpp:289:22: warning: missing initializer for member ‘KeyData::profile’ [-Wmissing-field-initializers]
     KeyData data = {0};
                      ^
customkeywidget.cpp:289:22: warning: missing initializer for member ‘KeyData::key_value’ [-Wmissing-field-initializers]
customkeywidget.cpp:289:22: warning: missing initializer for member ‘KeyData::macro_type’ [-Wmissing-field-initializers]
customkeywidget.cpp:289:22: warning: missing initializer for member ‘KeyData::macro_value’ [-Wmissing-field-initializers]
customkeywidget.cpp:289:22: warning: missing initializer for member ‘KeyData::macro_value1’ [-Wmissing-field-initializers]
customkeywidget.cpp:289:22: warning: missing initializer for member ‘KeyData::macro_value2’ [-Wmissing-field-initializers]
customkeywidget.cpp:289:22: warning: missing initializer for member ‘KeyData::macro_name’ [-Wmissing-field-initializers]
customkeywidget.cpp: At global scope:
customkeywidget.cpp:517:48: warning: unused parameter ‘event’ [-Wunused-parameter]
 void CustomKeyWidget::keyPressEvent(QKeyEvent *event)
                                                ^
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o custombutton.o custombutton.cpp
custombutton.cpp:47:41: warning: unused parameter ‘e’ [-Wunused-parameter]
 void tabButton::paintEvent(QPaintEvent* e)
                                         ^
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o customknobwidget.o customknobwidget.cpp
customknobwidget.cpp: In member function ‘void CustomKnobWidget::on_pushButton_save_clicked()’:
customknobwidget.cpp:103:22: warning: missing initializer for member ‘KeyData::profile’ [-Wmissing-field-initializers]
     KeyData data = {0};
                      ^
customknobwidget.cpp:103:22: warning: missing initializer for member ‘KeyData::key_value’ [-Wmissing-field-initializers]
customknobwidget.cpp:103:22: warning: missing initializer for member ‘KeyData::macro_type’ [-Wmissing-field-initializers]
customknobwidget.cpp:103:22: warning: missing initializer for member ‘KeyData::macro_value’ [-Wmissing-field-initializers]
customknobwidget.cpp:103:22: warning: missing initializer for member ‘KeyData::macro_value1’ [-Wmissing-field-initializers]
customknobwidget.cpp:103:22: warning: missing initializer for member ‘KeyData::macro_value2’ [-Wmissing-field-initializers]
customknobwidget.cpp:103:22: warning: missing initializer for member ‘KeyData::macro_name’ [-Wmissing-field-initializers]
customknobwidget.cpp: In member function ‘void CustomKnobWidget::on_pushButton_cancel_clicked()’:
customknobwidget.cpp:268:22: warning: missing initializer for member ‘KeyData::profile’ [-Wmissing-field-initializers]
     KeyData data = {0};
                      ^
customknobwidget.cpp:268:22: warning: missing initializer for member ‘KeyData::key_value’ [-Wmissing-field-initializers]
customknobwidget.cpp:268:22: warning: missing initializer for member ‘KeyData::macro_type’ [-Wmissing-field-initializers]
customknobwidget.cpp:268:22: warning: missing initializer for member ‘KeyData::macro_value’ [-Wmissing-field-initializers]
customknobwidget.cpp:268:22: warning: missing initializer for member ‘KeyData::macro_value1’ [-Wmissing-field-initializers]
customknobwidget.cpp:268:22: warning: missing initializer for member ‘KeyData::macro_value2’ [-Wmissing-field-initializers]
customknobwidget.cpp:268:22: warning: missing initializer for member ‘KeyData::macro_name’ [-Wmissing-field-initializers]
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o keysetwidget.o keysetwidget.cpp
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o radilightwidget.o radilightwidget.cpp
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o devicecomm.o devicecomm.cpp
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSql -I/usr/include/qt4 -I. -I. -o devicedb.o devicedb.cpp
In file included from devicedb.h:9:0,
                 from devicedb.cpp:1:
keyboarddata.h:76:64: error: ‘uint8_t’ has not been declared
 void hid_keyRemapData(int src_key, int dst_key, int kb_layout, uint8_t* kb_data
                                                                ^
keyboarddata.h:79:98: error: ‘uint8_t’ has not been declared
 _key, int dst_syskey1, int dst_syskey2, int dst_key, int kb_layout, uint8_t* kb
                                                                     ^
In file included from devicedb.h:9:0,
                 from devicedb.cpp:1:
keyboarddata.h:82:94: error: ‘uint8_t’ has not been declared
 _key,  int play_times, QVector<KeyMacro*> &vec_data, int kb_layout, uint8_t* kb
                                                                     ^
keyboarddata.h:85:55: error: ‘uint8_t’ has not been declared
 void hid_keyDefaultValue(int src_key,  int kb_layout, uint8_t* kb_data, int kb_
                                                       ^
keyboarddata.h:88:55: error: ‘uint8_t’ has not been declared
 void hid_disablekeyValue(int src_key,  int kb_layout, uint8_t* kb_data, int kb_
                                                       ^
keyboarddata.h:91:37: error: ‘uint8_t’ has not been declared
 void hid_getRadiData(RadiData data, uint8_t* kb_data);
                                     ^
keyboarddata.h:94:67: error: ‘uint8_t’ has not been declared
 void hid_getKeyRGBData(QVector<RGBData*> &vecdata, int kb_layout, uint8_t* kb_d
                                                                   ^
keyboarddata.h:97:43: error: ‘uint8_t’ has not been declared
 void hid_getKeyboardReportRate(int value, uint8_t* kb_data);
                                           ^
keyboarddata.h:100:40: error: ‘uint8_t’ has not been declared
 void hid_getUSBModeSleepTime(int time, uint8_t* kb_data);
                                        ^
keyboarddata.h:103:74: error: ‘uint8_t’ has not been declared
 hid_getWirelessModeSleepTime(int backlight_time, int wireless_time, uint8_t* kb
                                                                     ^
keyboarddata.h:106:27: error: variable or field ‘hid_getResetKeyboard’ declared void
 void hid_getResetKeyboard(uint8_t* kb_data);
                           ^
keyboarddata.h:106:27: error: ‘uint8_t’ was not declared in this scope
keyboarddata.h:106:36: error: ‘kb_data’ was not declared in this scope
 void hid_getResetKeyboard(uint8_t* kb_data);
                                    ^
Makefile:424: recipe for target 'devicedb.o' failed
make: *** [devicedb.o] Error 1
```

Every helps will be greatly appreciated!

--- a/README.md
+++ b/README.md
@@ -11,8 +11,24 @@ libusb: https://sourceforge.net/projects/libusb/
 Installation
 --
 
-Arch Linux: Package available through [AUR](https://aur.archlinux.org/packages/drevo-power-console-git/)
+#### Arch Linux:
+Package available through [AUR](https://aur.archlinux.org/packages/drevo-power-console-git/)
 
+#### Building Manually
+
+###### Debian/Ubuntu package requirements
+```bash
+sudo apt install build-essential qt5-default libusb-1.0-0-dev
+```
+
+###### run:
+```bash
+cd dpc_linux
+qmake -makefile DrevoPowerConsole.pro
+make
+```
+
+###### Udev Fix
 To be able to run the program with non root access you will need to copy the `udev` rule to your installation:
 
 ```bash

